app/src/androidTest/java/ch/epfl/sdp/blindly/weather/WeatherActivityTest.kt

```diff
@@ -0,0 +1,197 @@
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @Inject
+    lateinit var weather: WeatherService
+    private var decorView: View? = null
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+        activityRule.scenario.onActivity { activity ->
+            activity.findViewById<SwipeRefreshLayout>(R.id.swiperefresh).isRefreshing = false
+            decorView = activity.getWindow().getDecorView()
+        }
+    }
+
+    @After
+    fun afterEach() {
+        Intents.release()
+    }
+
+
+    @Test
+    fun dragToRefreshFetchesNewDataAndDisplayThem() {
+        // Override default result
+        `when`(weather.nextWeek(any(), any(), any()))
+            .then(FakeWeatherServiceModule.answerResult(WEEK_WEATHER_2))
+
+        performRefresh()
+
+        verifyMockCalledAgainAndViewUpdated()
+
+    }
+
+    @Test
+    fun buttonRefreshFetchesNewDataAndDisplayThem() {
+        // Override default result
+        `when`(weather.nextWeek(any(), any(), any()))
+            .then(FakeWeatherServiceModule.answerResult(WEEK_WEATHER_2))
+
+        // Open menu
+        Espresso.openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getInstrumentation().targetContext);
+        // We can't use withId in action bar
+        // https://stackoverflow.com/a/24743493
+        onView(withText(R.string.menu_refresh)).perform(click())
+
+        verifyMockCalledAgainAndViewUpdated()
+
+    }
+
+    @Test
+    fun hideRefreshAfterResult() {
+        `when`(weather.nextWeek(any(), any(), any()))
+            .then(FakeWeatherServiceModule.answerResult(WEEK_WEATHER_2))
+
+        performRefresh()
+        verifyMockCalledAgainAndViewUpdated()
+        assertNotRefreshing()
+    }
+    @Test
+    fun unitsAreCorrectlyDisplayed() {
+        onView(withId(R.id.weather_day_1_day))
+            .check(
+                matches(
+                    withText(
+                        // We only roughly check, it's up to the implementation to choose
+                        // the precision of the display
+                        containsString("°C")
+                    )
+                )
+            )
+        `when`(weather.nextWeek(any(), any(), any()))
+            .then(FakeWeatherServiceModule.answerResult(WEEK_WEATHER_FAHRENHEIT))
+        performRefresh()
+
+        onView(withId(R.id.weather_day_1_day))
+            .check(
+                matches(
+                    withText(
+                        // We only roughly check, it's up to the implementation to choose
+                        // the precision of the display
+                        containsString("°F")
+                    )
+                )
+            )
+        assertNotRefreshing()
+    }
+    @Test
+    fun hideRefreshAfterFailure() {
+        `when`(weather.nextWeek(any(), any(), any()))
+            .then((Answer<Unit> {
+                val callback: WeatherService.WeatherResultCallback = it.getArgument(
+                    2,
```

Comment:
I was wondering about this 2 const, it there a way to make it a bit more explicit?